### PR TITLE
allow async getJSON (& adapt saveJSON)

### PR DESCRIPTION
**Pull request by flying-sheep**
why not go the whole way with promises?

this way we don’t force people (e.g. me 😛) to use sync variants of IO functions.

### CODE REVIEW

File: src/saveJSON.js [hunk 1]
```diff
@@ -1,5 +1,10 @@
-import { writeFileSync } from 'fs';
+import { writeFile } from 'fs';
 
 export default function saveJSON(cssFile, json) {
-  writeFileSync(`${ cssFile }.json`, JSON.stringify(json));
+  return new Promise((resolve, reject) => {
+    writeFile(
+      `${ cssFile }.json`,
+      JSON.stringify(json),
+      e => e ? reject(e) : resolve());
```
Comment (outpunk) [inline src/saveJSON.js]: It can be useful to resolve the promise with the received json. Could you please make this?

**Comment by outpunk** [comment]
@flying-sheep, great! I'll merge this today or tomorrow (on my phone right now). Thank you!

**Comment by outpunk** [comment]
@flying-sheep could you also fix the failing tests please?

**Comment by flying-sheep** [comment]
OK, done!

**Comment by outpunk** [comment]
@flying-sheep published

**Comment by flying-sheep** [comment]
thank you!

**Pull request merged**